main.go

```diff
@@ -45,7 +47,6 @@ var (
 	discoveryInterval = flag.Duration("discovery-interval", 30*time.Second, "How often to refresh discovery cache to pick up newly-installed resources")
 	informerRelist    = flag.Duration("cache-flush-interval", 30*time.Minute, "How often to flush local caches and relist objects from the API server")
 	debugAddr         = flag.String("debug-addr", ":9999", "The address to bind the debug http endpoints")
-	clientConfigPath  = flag.String("client-config-path", "", "Path to kubeconfig file (same format as used by kubectl); if not specified, use in-cluster config")
```

Comment:
The `controller-runtime` package will look for an optional `--kubeconfig` so this one should not be needed anymore. Changing the flag name is a breaking change and we can do one of two things
- Keep the flag and somehow adapt controller-runtime to use this one
- Document the breaking change in the release notes

Reply:
if its possible I would took non-breaking change approach, like marking flag as deprecated for now and pass it to controller-runtime, but would need to see whole PR, so please hold with adjusting that for now